pkg/cloudprovider/providers/oci/instances.go

```diff
@@ -101,7 +102,28 @@ func (cp *CloudProvider) NodeAddressesByProviderID(ctx context.Context, provider
 	if err != nil {
 		return nil, errors.Wrap(err, "GetPrimaryVNICForInstance")
 	}
-	return extractNodeAddressesFromVNIC(vnic)
+	vnicAddresses, err := extractNodeAddressesFromVNIC(vnic)
+	if err != nil {
+		return nil, err
+	}
+	addresses = append(addresses, vnicAddresses...)
+	hostname := vnic.HostnameLabel
+	if *hostname == "" {
+		cp.logger.With("instanceID", providerID).Debug("hostname not set, instance won't be resolvable using using the Internet and VCN Resolver")
+		return addresses, nil
+	}
+	subnet, err := cp.client.Networking().GetSubnet(ctx, *vnic.SubnetId)
```

Comment:
If `DnsLabel` attribute is not specified while creating VCN or subnet, the Internet & VCN resolver will not be able to resolve hostnames of instances in this VCN or subnet(https://docs.cloud.oracle.com/iaas/api/#/en/iaas/20160918/datatypes/CreateVcnDetails). Shall we add checks for these before forming `fqdn`.